dev/devicelab/bin/run.dart

```diff
@@ -13,129 +13,274 @@ import 'package:flutter_devicelab/framework/task_result.dart';
 import 'package:flutter_devicelab/framework/utils.dart';
 import 'package:path/path.dart' as path;
 
-late ArgResults args;
-
-List<String> _taskNames = <String>[];
-
-/// The device-id to run test on.
-String? deviceId;
+/// Runs tasks.
+///
+/// The tasks are chosen depending on the command-line options.
+Future<void> main(List<String> rawArgs) async {
+  // This is populated by a callback in the ArgParser.
+  final List<String> taskNames = <String>[];
+
+  final ArgParser argParser = ArgParser()
+    ..addMultiOption(
```

Comment:
nit: I prefer seeing these defined somewhere outside of main, or even in its own file - although I'm not sure there's any really great solution here, since that ends up meaning a reader might not have a clear idea of what arguments are defined and what their types/expected values are. 

Reason for preferring outside of main is that this is 100+ lines of configuration/documentation that I have to scroll through to get to the meat of what main is doing.

Reply:
totally reasonable; I'll move it back into its own function but with more of a functional style rather than the global variables style from before.